tools/serverpod_cli/lib/src/generator/dart/library_generators/model_library_generator.dart

```diff
@@ -1640,7 +1643,16 @@ class SerializableModelLibraryGenerator {
           classDefinition.subDirParts,
         ),
         ..._buildModelTableClassManyRelationGetters(fields, classDefinition),
-        _buildModelTableClassColumnGetter(fields),
+        _buildModelTableClassColumnGetter(
+          serializedFields,
+          name: 'columns',
+        ),
+        if (serializedFields.any((f) => f.hiddenSerializableField(serverCode)))
```

Comment:
Checks if overwrite needed..

Perhaps introduce `hiddenFields = serializedFields.where(...).toSet()` and test for empty.